### PR TITLE
Proxy Handler

### PR DESCRIPTION
**Pull request by plantver**
The proxy handler should be assigned to path `/`
i.e. the config should be 
`
path / ProxyHandler { 
	host ucla.edu; 
	port 80;
} 
`

Changes are made to server_config.cc to enable proxy handler.

Problem:
URI not correctly mapped
e.g. `localhost:9999/echo` will be separated into `path \` and  `file echo` resulting in ProxyHandler being called.

TODO for Team12:

- write tests for proxy handler

### CODE REVIEW

**Comment by binerys** [comment]
Hi, thank you for pointing the issue with assigning paths to: "/" 
I just updated our master branch to resolve that issue so now request handlers can be assigned to the "/" path. I saw that you made a minor edit to the one of the functions, but please let me know if the update I pushed works with your code as well and passes our integration test. 

I'll review the rest of the code tomorrow. 

Thanks!

**Comment by plantver** [comment]
More features added to Proxy Handler to deal with 302.
Tests will be added later.

File: src/request_handler.h [hunk 1]
```diff
@@ -101,4 +102,15 @@ class StatusHandler : public RequestHandler {
 
 REGISTER_REQUEST_HANDLER(StatusHandler);
 
+class ProxyHandler : public RequestHandler {
+  public:
+  virtual Status Init(const std::string& uri_prefix, const NginxConfig& config);
```
Comment (natalieethell) [inline src/request_handler.h]: maybe just indent these function declarations for aesthetic purposes :)

Comment (plantver) [inline src/request_handler.h]: sure

**Comment by plantver** [comment]
Merged with latest commit.

**Comment by binerys** [comment]
Hi, 

Can you move the proxy integration test into the test folder? 

Also, I'm not sure why the build is failing as I just ran the integration test locally with your branch and it works fine. Hopefully updating this branch and trying test again will yield a success.

**Comment by binerys** [comment]
@plantver @liuchang2872  It's been merged in!

**Pull request merged**